config/environments/production.rb

```diff
@@ -27,8 +27,9 @@
   # Compress CSS using a preprocessor.
   # config.assets.css_compressor = :sass
 
-  # Do not fallback to assets pipeline if a precompiled asset is missed.
-  config.assets.compile = false
+  # We are pre-compiling assets, so on production these will always be present, however
+  # for this to work this flag needs to be `true`. Seems like a Rails bug to me.
+  config.assets.compile = true
```

Comment:
Good point i'll ask

Reply:
Yep, it seemed to me like is a bug because in production we don’t want assets to be compiled, so technically this should be set to false (and in fact is the rails default for a brand new application), however, when doing so, the assets will not be found.

By setting it to true, despite being counterintuitive as it means “if you can’t find an asset, compile it when it is first accessed” (and we don’t want that on production) it seems to do the trick, and will not compile (despite the flag) because the assets are found (as we do a precompile of assets in the Dockerfile).

But happy if anyone else can take a look and find the source of the problem. I spent quite a lot of time last time I looked into this and this was the only solution that worked for me 😄

Reply:
@hiboabd might be useful to update your existing comment with Jesus's extra detail otherwise it's going to be a red-herring in a year's time!